src/matsim_io/scripts.py

```diff
@@ -59,3 +59,36 @@ def _update_dashboard_title(dashboard: Path, title: str) -> None:
     data = yaml.safe_load(dashboard.read_text(encoding="utf-8"))
     data["header"]["title"] = title
     dashboard.write_text(yaml.dump(data, sort_keys=False), encoding="utf-8")
+
+
+def append_breakpoints_to_congestion_map(output_dir: str) -> None:
+    """
+    Append breakpoints to the congestion map.
+    """
+    file_path = MATSIM_DATA_DIR / output_dir / "dashboard-2.yaml"
+    lines = file_path.read_text(encoding="utf-8").splitlines(keepends=True)
+    updated_lines = []
+    inside_congestion_block = False
+
+    for line in lines:
+        updated_lines.append(line)
+
+        # Detect start of congestion_map entry
+        if line.strip() == "- type: xytime":
+            inside_congestion_block = True
+
+        # Find the right place to insert: after the last key in the block
+        elif inside_congestion_block and (
+            line.startswith("  ") and not line.startswith("    ")
+        ):
+            # This line is not part of the current item anymore
+            updated_lines.insert(
+                -1, "    breakpoints: [0.1,0.2,0.3,0.4,0.5,0.6,0.7,0.8,0.9]\n"
+            )
+            inside_congestion_block = False
+
+    # Edge case: if it's the last block and never exited
+    if inside_congestion_block:
+        updated_lines.append("    breakpoints: [0.1,0.2,0.3,0.4,0.5,0.6,0.7,0.8,0.9]\n")
+
+    file_path.write_text("".join(updated_lines), encoding="utf-8")
```

Comment:
If you want, you can use PyYAML instead, to avoid a lot of the manual work worrying about whitespaces and correct insertion placement:

```suggestion
    data = yaml.safe_load(file_path.read_text(encoding="utf-8"))
    xytime_plot = data["layout"]["congestion_map"][0]
    xytime_plot["breakpoints"] = [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9]
    file_path.write_text(yaml.dump(data, sort_keys=False), encoding="utf-8")
```

Reply:
A sanity check might also be a good idea:

```suggestion
    data = yaml.safe_load(file_path.read_text(encoding="utf-8"))
    if "congestion_map" not in data["layout"]:
        logging.error("No congestion map found in the dashboard file.")
        return
    xytime_plot = data["layout"]["congestion_map"][0]
    if xytime_plot["type"] != "xytime":
        logging.error("Could not find XYTime plot in the dashboard file.")
        return
    xytime_plot["breakpoints"] = [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9]
    file_path.write_text(yaml.dump(data, sort_keys=False), encoding="utf-8")
```